src/App.tsx

```diff
@@ -31,62 +31,98 @@ import ProtectedRoute from './components/ProtectedRoute/ProtectedRoute';
-      <Redirect from='/volunteer/signup' to='/volunteer' />
-      <ProtectedRoute exact path='/volunteer/:requestId' component={VolunteerDetail} />
-      <Redirect exact from='/profile' to='/profile/me/' />
-      <Route exact path='/profile/:userId/' component={UserProfile} />
-    </Switch>
-  )
-  const public_routes = (
+};
+
+const VolunteerRoutes = () => {
+  return (
     <Switch>
-      <Route exact path='/' component={Home} />
-      <Route exact path='/get-help' component={GetHelp} />
-      <Route exact path='/get-involved' component={GetInvolved} />
-      <Route exact path='/volunteer/signup' component={SignUp} />
-      <Redirect from="/admin" to="/admin/" />
+      <ProtectedRoute exact path="/volunteer" component={VolunteerList} />
+      <ProtectedRoute
+        exact
+        path="/volunteer/my-support"
+        component={MyVolunteerList}
+      />
+      <ProtectedRoute
+        exact
+        path="/volunteer/my-support/:volunteerId"
+        component={MyVolunteerDetail}
+      />
+      <Redirect from="/volunteer/signup" to="/volunteer" />
+      <ProtectedRoute
+        exact
+        path="/volunteer/:requestId"
+        component={VolunteerDetail}
+      />
+      <Redirect exact from="/profile" to="/profile/me/" />
+      <Route exact path="/profile/:userId/" component={UserProfile} />
     </Switch>
   );
+};
+
+const App = ({
+  isAuthenticated,
+  authCheckLoginState,
+  is_requester,
+  is_volunteer,
+}) => {
+  const token = localStorage.getItem('access');
+
+  useEffect(() => {
+    authCheckLoginState();
+  }, [token, authCheckLoginState]);
 
   return (
     <Layout>
-      {public_routes}
+      <Switch>
+        <Route exact path="/" component={Home} />
+        <Route exact path="/get-help" component={GetHelp} />
+        <Route exact path="/get-involved" component={GetInvolved} />
+        <Route exact path="/volunteer/signup" component={SignUp} />
+        <Redirect from="/admin" to="/admin/" />
+      </Switch>
       <Route exact path="/logout" component={Logout} />
-      <Route exact path='/password-reset' component={ResetPassword} />
-      <Route exact path="/password-reset-confirm/:uid/:token" component={ConfirmPassword}></Route>
-      <Route exact path='/email-reset' component={ResetEmail} />
-      <Route exact path="/email-reset-confirm/:uid/:token" component={ConfirmEmail}></Route>
-      {is_requester && isAuthenticated ? requester_routes : null}
-      {is_volunteer && isAuthenticated ? volunteer_routes : null}
+      <Route exact path="/password-reset" component={ResetPassword} />
+      <Route
+        exact
+        path="/password-reset-confirm/:uid/:token"
+        component={ConfirmPassword}
+      />
+      <Route exact path="/email-reset" component={ResetEmail} />
+      <Route
+        exact
+        path="/email-reset-confirm/:uid/:token"
+        component={ConfirmEmail}
+      />
+      <RequesterRoutes />
+      <VolunteerRoutes />
```

Comment:
these are the primary changes in this PR.

1. I don't quite understand the logic behind restricting possible routes by someone's role, since those pages/containers are authenticated when visited. maybe I am missing something.
2. change these from constant vars to jsx function components. my primary reason to do this is for standardization/better understanding while reading this component. I don't know if that's successful, but since they rendering jsx, using the standard react best-practices seems to make sense

Reply:
1. Since we want to separate requester and volunteer. We don't want volunteer to make requests (if they only have volunteer role) and we also don't want requester to go to volunteer routes and make volunteer-actions. Remove isAuthenticated is great because ProtectedRoute is for authenticated users .
```suggestion
      {is_requester && <RequesterRoutes />}
      {is_volunteer && <VolunteerRoutes />}
```

2. This is great. Should we make a PublicRoutes component to be clearer?

3. Also in ProtectedRoute Component should change render to
`render{props => isAuthenticated && !loading && <Component {...props} />}`

Reply:
> Since we want to separate requester and volunteer. We don't want volunteer to make requests (if they only have volunteer role) and we also don't want requester to go to volunteer routes and make volunteer-actions. Remove isAuthenticated is great because ProtectedRoute is for authenticated users .

in my experience, it's a bit unorthodox to define routes based on a user's current role. I agree that we do not want Requesters to see Volunteer pages, but the logic to block them from seeing data is part of the individual routes/components. if a user is logged in as a Volunteer, and they go to a Requester route, they will be blocked by the Requester container, correct? and the same thing the other way?

Reply:
Sounds great. So we should do role check in individual components?